### PR TITLE
4 feature 4 search stock

### PR DESCRIPTION
**Pull request by noel-toms**
I moved a lot of functionality such as the button logic and loading of the stock information from the GUI to the use case interactor to adhere to the clean architecture rules. I also fixed the issue of handling a wrong input, it now displays a message to the user. I just created a method in my use case interactor to check stock symbol validity. I also added a loading message that displays while the search stock is being loaded. I did this as I noticed that the API takes some time to get all the data, and the user is presented with a blank screen while they wait.

### CODE REVIEW

File: src/main/java/BuyStockUseCase/PortfolioPresenter.java [hunk 1]
```diff
@@ -39,6 +42,12 @@ public void onBack() {
     }
     public void onSearch() {
         String symbol = view.getSearchField();
+        try{
+            new ViewStockController(symbol).stockIsValid();
+        }catch (Exception e){
+            JOptionPane.showMessageDialog(null, "Invalid stock symbol: "+symbol);
```
Comment (inaam10-tech) [inline src/main/java/BuyStockUseCase/PortfolioPresenter.java:48]: I think you could refactor your code around here and place it in a different file

File: src/main/java/SearchStockUseCase/ViewStockUseCaseInteractor.java [hunk 2]
```diff
@@ -1,23 +1,77 @@
...
+    }
 
+    public void isValidStock() throws Exception {
+        //Checking is API throws error or not
+        try{
+            new StockAPIGateway().getPrice(new StockAPIRequest(this.stockSymbol));
+        }catch (IOException e){
+            throw new Exception("Invalid stock symbol");
+        }
+    }
```
Comment (inaam10-tech) [inline src/main/java/SearchStockUseCase/ViewStockUseCaseInteractor.java:34]: could serve to be useful adding Javadocs

**Review by inaam10-tech — CHANGES_REQUESTED** [review]
Added a few comments that you can consider (mostly consider adding Javadocs and )and I also notice an issue with the checks. I think it may be related to some merging conflict.

**Review by Spaghetti-12 — DISMISSED** [review]
Looks like the autograder is upset about something. here's the error message:

  /home/runner/work/course-project-Team-Stonks/course-project-Team-Stonks/src/test/java/SearchStockTest/SearchStockInteractorTest.java:13: error: constructor ViewStockUseCaseInteractor in class ViewStockUseCaseInteractor cannot be applied to given types;
          interactor = new ViewStockUseCaseInteractor();
                       ^
    required: String
    found: no arguments
    reason: actual and formal argument lists differ in length
  
  /home/runner/work/course-project-Team-Stonks/course-project-Team-Stonks/src/test/java/SearchStockTest/SearchStockInteractorTest.java:22: error: method searchStock in class ViewStockUseCaseInteractor cannot be applied to given types;
              boolean isValid = interactor.searchStock(incorrectSymbol);
                                          ^
    required: no arguments
    found: String
    reason: actual and formal argument lists differ in length

File: src/main/java/BuyStockUseCase/PortfolioPresenter.java (hunk 1 above)
Comment (eladperlman) [reply src/main/java/BuyStockUseCase/PortfolioPresenter.java:48]: yea this code should be put into a function in the GUI class called presentInvalidStockMessage and the presenter should call it through the interface the GUI implements

Comment (noel-toms) [reply src/main/java/BuyStockUseCase/PortfolioPresenter.java]: Thanks for the feedback, I have moved some logic into a function in the GUI.

File: src/main/java/SearchStockUseCase/ViewStockUseCaseInteractor.java (hunk 2 above)
Comment (noel-toms) [reply src/main/java/SearchStockUseCase/ViewStockUseCaseInteractor.java]: I have now added some documentation in the use case interactor

**Review by inaam10-tech — APPROVED** [review]
Great job. I see you refactored the code.

**Review by jackzxliu — APPROVED** [review]

**Review by Peaceful36 — APPROVED** [review]
Nice job

**Pull request merged**